{"text": "2025-02-03 02:54:00 - ERROR    - utils - Error creating DB and tables\nTraceback (most recent call last):\n\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 146, in __init__\n    self._dbapi_connection = engine.raw_connection()\n    |                        |      -> <function Engine.raw_connection at 0xffff8079e5c0>\n    |                        -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n    -> <sqlalchemy.engine.base.Connection object at 0xffff4ffe4650>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 3298, in raw_connection\n    return self.pool.connect()\n           |    |    -> <function Pool.connect at 0xffff80e028e0>\n           |    -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 449, in connect\n    return _ConnectionFairy._checkout(self)\n           |                |         -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           |                -> <classmethod(<function _ConnectionFairy._checkout at 0xffff80df8a40>)>\n           -> <class 'sqlalchemy.pool.base._ConnectionFairy'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 1263, in _checkout\n    fairy = _ConnectionRecord.checkout(pool)\n            |                 |        -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n            |                 -> <classmethod(<function _ConnectionRecord.checkout at 0xffff80e03560>)>\n            -> <class 'sqlalchemy.pool.base._ConnectionRecord'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 712, in checkout\n    rec = pool._do_get()\n          |    -> <function NullPool._do_get at 0xffff80e48860>\n          -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/impl.py\", line 308, in _do_get\n    return self._create_connection()\n           |    -> <function Pool._create_connection at 0xffff80e02660>\n           -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 390, in _create_connection\n    return _ConnectionRecord(self)\n           |                 -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           -> <class 'sqlalchemy.pool.base._ConnectionRecord'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 674, in __init__\n    self.__connect()\n    -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 900, in __connect\n    with util.safe_reraise():\n         |    -> <class 'sqlalchemy.util.langhelpers.safe_reraise'>\n         -> <module 'sqlalchemy.util' from '/usr/local/lib/python3.12/site-packages/sqlalchemy/util/__init__.py'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n          |         |              -> <traceback object at 0xffff7ca7f040>\n          |         -> <method 'with_traceback' of 'BaseException' objects>\n          -> OperationalError('unable to open database file')\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 896, in __connect\n    self.dbapi_connection = connection = pool._invoke_creator(self)\n    |    |                               |    |               -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n    |    |                               |    -> <function create_engine.<locals>.connect at 0xffff7cab42c0>\n    |    |                               -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n    |    -> <member 'dbapi_connection' of '_ConnectionRecord' objects>\n    -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/create.py\", line 646, in connect\n    return dialect.connect(*cargs, **cparams)\n           |       |        |        -> {'check_same_thread': False, 'timeout': 20}\n           |       |        -> ['/usr/local/lib/python3.12/site-packages/langflow/langflow.db']\n           |       -> <function DefaultDialect.connect at 0xffff809265c0>\n           -> <sqlalchemy.dialects.sqlite.aiosqlite.SQLiteDialect_aiosqlite object at 0xffff50afa600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/default.py\", line 622, in connect\n    return self.loaded_dbapi.connect(*cargs, **cparams)\n           |    |            |        |        -> {'check_same_thread': False, 'timeout': 20}\n           |    |            |        -> ('/usr/local/lib/python3.12/site-packages/langflow/langflow.db',)\n           |    |            -> <function AsyncAdapt_aiosqlite_dbapi.connect at 0xffff53e918a0>\n           |    -> <sqlalchemy.dialects.sqlite.aiosqlite.AsyncAdapt_aiosqlite_dbapi object at 0xffff512424b0>\n           -> <sqlalchemy.dialects.sqlite.aiosqlite.SQLiteDialect_aiosqlite object at 0xffff50afa600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/dialects/sqlite/aiosqlite.py\", line 355, in connect\n    await_only(connection),\n    |          -> <Connection(Thread-1, stopped daemon 281472138539392)>\n    -> <function await_only at 0xffff813931a0>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py\", line 132, in await_only\n    return current.parent.switch(awaitable)  # type: ignore[no-any-return,attr-defined] # noqa: E501\n           |       |             -> <Connection(Thread-1, stopped daemon 281472138539392)>\n           |       -> <attribute 'parent' of 'greenlet.greenlet' objects>\n           -> <_AsyncIoGreenlet object at 0xffff7ca7f400 (otid=0xffff813cb540) dead>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py\", line 196, in greenlet_spawn\n    value = await result\n                  -> <Connection(Thread-1, stopped daemon 281472138539392)>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 140, in _connect\n    self._connection = await future\n    |    |                   -> <Future finished exception=OperationalError('unable to open database file')>\n    |    -> None\n    -> <Connection(Thread-1, stopped daemon 281472138539392)>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 115, in run\n    result = function()\n             -> <function connect.<locals>.connector at 0xffff7ca58720>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 392, in connector\n    return sqlite3.connect(loc, **kwargs)\n           |       |       |      -> {'check_same_thread': False, 'timeout': 20}\n           |       |       -> '/usr/local/lib/python3.12/site-packages/langflow/langflow.db'\n           |       -> <built-in function connect>\n           -> <module 'sqlite3' from '/usr/local/lib/python3.12/sqlite3/__init__.py'>\n\nsqlite3.OperationalError: unable to open database file\n\n\nThe above exception was the direct cause of the following exception:\n\n\nTraceback (most recent call last):\n\n  File \"/usr/local/bin/langflow\", line 10, in <module>\n    sys.exit(main())\n    |   |    -> <function main at 0xffff50bebec0>\n    |   -> <built-in function exit>\n    -> <module 'sys' (built-in)>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/__main__.py\", line 618, in main\n    app()\n    -> <typer.main.Typer object at 0xffff50d483b0>\n  File \"/usr/local/lib/python3.12/site-packages/typer/main.py\", line 323, in __call__\n    return get_command(self)(*args, **kwargs)\n           |           |      |       -> {}\n           |           |      -> ()\n           |           -> <typer.main.Typer object at 0xffff50d483b0>\n           -> <function get_command at 0xffff8162dda0>\n  File \"/usr/local/lib/python3.12/site-packages/click/core.py\", line 1161, in __call__\n    return self.main(*args, **kwargs)\n           |    |     |       -> {}\n           |    |     -> ()\n           |    -> <function TyperGroup.main at 0xffff8162c400>\n           -> <TyperGroup >\n  File \"/usr/local/lib/python3.12/site-packages/typer/core.py\", line 743, in main\n    return _main(\n           -> <function _main at 0xffff81627560>\n  File \"/usr/local/lib/python3.12/site-packages/typer/core.py\", line 198, in _main\n    rv = self.invoke(ctx)\n         |    |      -> <click.core.Context object at 0xffff50d305c0>\n         |    -> <function MultiCommand.invoke at 0xffff8287cf40>\n         -> <TyperGroup >\n  File \"/usr/local/lib/python3.12/site-packages/click/core.py\", line 1697, in invoke\n    return _process_result(sub_ctx.command.invoke(sub_ctx))\n           |               |       |       |      -> <click.core.Context object at 0xffff800dbe60>\n           |               |       |       -> <function Command.invoke at 0xffff8287c900>\n           |               |       -> <TyperCommand run>\n           |               -> <click.core.Context object at 0xffff800dbe60>\n           -> <function MultiCommand.invoke.<locals>._process_result at 0xffff50aa04a0>\n  File \"/usr/local/lib/python3.12/site-packages/click/core.py\", line 1443, in invoke\n    return ctx.invoke(self.callback, **ctx.params)\n           |   |      |    |           |   -> {'host': '0.0.0.0', 'port': 7860, 'log_level': 'info', 'workers': None, 'worker_timeout': None, 'components_path': PosixPath(...\n           |   |      |    |           -> <click.core.Context object at 0xffff800dbe60>\n           |   |      |    -> <function run at 0xffff83609f80>\n           |   |      -> <TyperCommand run>\n           |   -> <function Context.invoke at 0xffff828d3240>\n           -> <click.core.Context object at 0xffff800dbe60>\n  File \"/usr/local/lib/python3.12/site-packages/click/core.py\", line 788, in invoke\n    return __callback(*args, **kwargs)\n                       |       -> {'host': '0.0.0.0', 'port': 7860, 'log_level': 'info', 'workers': None, 'worker_timeout': None, 'components_path': PosixPath(...\n                       -> ()\n  File \"/usr/local/lib/python3.12/site-packages/typer/main.py\", line 698, in wrapper\n    return callback(**use_params)\n           |          -> {'host': '0.0.0.0', 'workers': None, 'worker_timeout': None, 'port': 7860, 'components_path': PosixPath('/usr/local/lib/pytho...\n           -> <function run at 0xffff50beb2e0>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/__main__.py\", line 226, in run\n    process = run_on_mac_or_linux(host, port, log_level, options, app)\n              |                   |     |     |          |        -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n              |                   |     |     |          -> {'bind': '0.0.0.0:7860', 'workers': 1, 'timeout': 300, 'worker_class': 'langflow.server.LangflowUvicornWorker', 'logger_class...\n              |                   |     |     -> 'info'\n              |                   |     -> 7860\n              |                   -> '0.0.0.0'\n              -> <function run_on_mac_or_linux at 0xffff50beb380>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/__main__.py\", line 262, in run_on_mac_or_linux\n    webapp_process.start()\n    |              -> <function BaseProcess.start at 0xffff813ae2a0>\n    -> <Process name='Process-1' parent=1 started>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/process.py\", line 121, in start\n    self._popen = self._Popen(self)\n    |    |        |    |      -> <Process name='Process-1' parent=1 started>\n    |    |        |    -> <staticmethod(<function Process._Popen at 0xffff812bdbc0>)>\n    |    |        -> <Process name='Process-1' parent=1 started>\n    |    -> None\n    -> <Process name='Process-1' parent=1 started>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/context.py\", line 224, in _Popen\n    return _default_context.get_context().Process._Popen(process_obj)\n           |                |                            -> <Process name='Process-1' parent=1 started>\n           |                -> <function DefaultContext.get_context at 0xffff812bdda0>\n           -> <multiprocess.context.DefaultContext object at 0xffff8161f9e0>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/context.py\", line 282, in _Popen\n    return Popen(process_obj)\n           |     -> <Process name='Process-1' parent=1 started>\n           -> <class 'multiprocess.popen_fork.Popen'>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/popen_fork.py\", line 19, in __init__\n    self._launch(process_obj)\n    |    |       -> <Process name='Process-1' parent=1 started>\n    |    -> <function Popen._launch at 0xffff5013a340>\n    -> <multiprocess.popen_fork.Popen object at 0xffff500f39e0>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/popen_fork.py\", line 71, in _launch\n    code = process_obj._bootstrap(parent_sentinel=child_r)\n           |           |                          -> 6\n           |           -> <function BaseProcess._bootstrap at 0xffff813aeca0>\n           -> <Process name='Process-1' parent=1 started>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/process.py\", line 314, in _bootstrap\n    self.run()\n    |    -> <function BaseProcess.run at 0xffff813ae200>\n    -> <Process name='Process-1' parent=1 started>\n  File \"/usr/local/lib/python3.12/site-packages/multiprocess/process.py\", line 108, in run\n    self._target(*self._args, **self._kwargs)\n    |    |        |    |        |    -> {}\n    |    |        |    |        -> <Process name='Process-1' parent=1 started>\n    |    |        |    -> ('0.0.0.0', 7860, 'info', {'bind': '0.0.0.0:7860', 'workers': 1, 'timeout': 300, 'worker_class': 'langflow.server.LangflowUvi...\n    |    |        -> <Process name='Process-1' parent=1 started>\n    |    -> <function run_langflow at 0xffff50beb920>\n    -> <Process name='Process-1' parent=1 started>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/__main__.py\", line 416, in run_langflow\n    server.run()\n    |      -> <function BaseApplication.run at 0xffff50010540>\n    -> <langflow.server.LangflowApplication object at 0xffff815c4050>\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/app/base.py\", line 71, in run\n    Arbiter(self).run()\n    |       -> <langflow.server.LangflowApplication object at 0xffff815c4050>\n    -> <class 'gunicorn.arbiter.Arbiter'>\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/arbiter.py\", line 210, in run\n    self.manage_workers()\n    |    -> <function Arbiter.manage_workers at 0xffff4ffcf380>\n    -> <gunicorn.arbiter.Arbiter object at 0xffff501d17c0>\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/arbiter.py\", line 570, in manage_workers\n    self.spawn_workers()\n    |    -> <function Arbiter.spawn_workers at 0xffff4ffcf4c0>\n    -> <gunicorn.arbiter.Arbiter object at 0xffff501d17c0>\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/arbiter.py\", line 641, in spawn_workers\n    self.spawn_worker()\n    |    -> <function Arbiter.spawn_worker at 0xffff4ffcf420>\n    -> <gunicorn.arbiter.Arbiter object at 0xffff501d17c0>\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/arbiter.py\", line 608, in spawn_worker\n    worker.init_process()\n    |      -> <function UvicornWorker.init_process at 0xffff50024860>\n    -> <langflow.server.LangflowUvicornWorker object at 0xffff4ff92150>\n  File \"/usr/local/lib/python3.12/site-packages/uvicorn/workers.py\", line 75, in init_process\n    super().init_process()\n  File \"/usr/local/lib/python3.12/site-packages/gunicorn/workers/base.py\", line 143, in init_process\n    self.run()\n    |    -> <function UvicornWorker.run at 0xffff4ffede40>\n    -> <langflow.server.LangflowUvicornWorker object at 0xffff4ff92150>\n  File \"/usr/local/lib/python3.12/site-packages/uvicorn/workers.py\", line 107, in run\n    return asyncio.run(self._serve())\n           |       |   |    -> <function LangflowUvicornWorker._serve at 0xffff50010c20>\n           |       |   -> <langflow.server.LangflowUvicornWorker object at 0xffff4ff92150>\n           |       -> <function run at 0xffff83708680>\n           -> <module 'asyncio' from '/usr/local/lib/python3.12/asyncio/__init__.py'>\n  File \"/usr/local/lib/python3.12/asyncio/runners.py\", line 194, in run\n    return runner.run(main)\n           |      |   -> <coroutine object LangflowUvicornWorker._serve at 0xffff501032a0>\n           |      -> <function Runner.run at 0xffff82940ea0>\n           -> <asyncio.runners.Runner object at 0xffff4ff92270>\n  File \"/usr/local/lib/python3.12/asyncio/runners.py\", line 118, in run\n    return self._loop.run_until_complete(task)\n           |    |     |                  -> <Task pending name='Task-1' coro=<LangflowUvicornWorker._serve() running at /usr/local/lib/python3.12/site-packages/langflow/...\n           |    |     -> <function BaseEventLoop.run_until_complete at 0xffff8293aac0>\n           |    -> <_UnixSelectorEventLoop running=True closed=False debug=False>\n           -> <asyncio.runners.Runner object at 0xffff4ff92270>\n  File \"/usr/local/lib/python3.12/asyncio/base_events.py\", line 674, in run_until_complete\n    self.run_forever()\n    |    -> <function BaseEventLoop.run_forever at 0xffff8293aa20>\n    -> <_UnixSelectorEventLoop running=True closed=False debug=False>\n  File \"/usr/local/lib/python3.12/asyncio/base_events.py\", line 641, in run_forever\n    self._run_once()\n    |    -> <function BaseEventLoop._run_once at 0xffff82940860>\n    -> <_UnixSelectorEventLoop running=True closed=False debug=False>\n  File \"/usr/local/lib/python3.12/asyncio/base_events.py\", line 1987, in _run_once\n    handle._run()\n    |      -> <function Handle._run at 0xffff82b12de0>\n    -> <Handle Task.task_wakeup(<Future finished result=None>)>\n  File \"/usr/local/lib/python3.12/asyncio/events.py\", line 88, in _run\n    self._context.run(self._callback, *self._args)\n    |    |            |    |           |    -> <member '_args' of 'Handle' objects>\n    |    |            |    |           -> <Handle Task.task_wakeup(<Future finished result=None>)>\n    |    |            |    -> <member '_callback' of 'Handle' objects>\n    |    |            -> <Handle Task.task_wakeup(<Future finished result=None>)>\n    |    -> <member '_context' of 'Handle' objects>\n    -> <Handle Task.task_wakeup(<Future finished result=None>)>\n  File \"/usr/local/lib/python3.12/site-packages/uvicorn/lifespan/on.py\", line 86, in main\n    await app(scope, self.receive, self.send)\n          |   |      |    |        |    -> <function LifespanOn.send at 0xffff7cadbe20>\n          |   |      |    |        -> <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>\n          |   |      |    -> <function LifespanOn.receive at 0xffff7cadbec0>\n          |   |      -> <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>\n          |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          -> <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0xffff4ff91040>\n  File \"/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py\", line 29, in __call__\n    return await self.app(scope, receive, send)\n                 |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                 |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                 |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n                 |    -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n                 -> <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0xffff4ff91040>\n  File \"/usr/local/lib/python3.12/site-packages/fastapi/applications.py\", line 1054, in __call__\n    await super().__call__(scope, receive, send)\n                           |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                           |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                           -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n  File \"/usr/local/lib/python3.12/site-packages/starlette/applications.py\", line 112, in __call__\n    await self.middleware_stack(scope, receive, send)\n          |    |                |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |                |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |                -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <starlette.middleware.errors.ServerErrorMiddleware object at 0xffff50001730>\n          -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/errors.py\", line 152, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <opentelemetry.instrumentation.asgi.OpenTelemetryMiddleware object at 0xffff500016a0>\n          -> <starlette.middleware.errors.ServerErrorMiddleware object at 0xffff50001730>\n  File \"/usr/local/lib/python3.12/site-packages/opentelemetry/instrumentation/asgi/__init__.py\", line 681, in __call__\n    return await self.app(scope, receive, send)\n                 |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                 |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n                 |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n                 |    -> <starlette.middleware.base.BaseHTTPMiddleware object at 0xffff50001670>\n                 -> <opentelemetry.instrumentation.asgi.OpenTelemetryMiddleware object at 0xffff500016a0>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py\", line 100, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <starlette.middleware.base.BaseHTTPMiddleware object at 0xffff500015e0>\n          -> <starlette.middleware.base.BaseHTTPMiddleware object at 0xffff50001670>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py\", line 100, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <langflow.main.JavaScriptMIMETypeMiddleware object at 0xffff500015b0>\n          -> <starlette.middleware.base.BaseHTTPMiddleware object at 0xffff500015e0>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py\", line 100, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <starlette.middleware.cors.CORSMiddleware object at 0xffff50001550>\n          -> <langflow.main.JavaScriptMIMETypeMiddleware object at 0xffff500015b0>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/cors.py\", line 77, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <langflow.middleware.ContentSizeLimitMiddleware object at 0xffff50001520>\n          -> <starlette.middleware.cors.CORSMiddleware object at 0xffff50001550>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/middleware.py\", line 55, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <starlette.middleware.exceptions.ExceptionMiddleware object at 0xffff500014f0>\n          -> <langflow.middleware.ContentSizeLimitMiddleware object at 0xffff50001520>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/middleware/exceptions.py\", line 48, in __call__\n    await self.app(scope, receive, send)\n          |    |   |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |   -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <fastapi.routing.APIRouter object at 0xffff50ab6e40>\n          -> <starlette.middleware.exceptions.ExceptionMiddleware object at 0xffff500014f0>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/routing.py\", line 715, in __call__\n    await self.middleware_stack(scope, receive, send)\n          |    |                |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |                |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |                -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <bound method Router.app of <fastapi.routing.APIRouter object at 0xffff50ab6e40>>\n          -> <fastapi.routing.APIRouter object at 0xffff50ab6e40>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/routing.py\", line 724, in app\n    await self.lifespan(scope, receive, send)\n          |    |        |      |        -> <bound method LifespanOn.send of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |        |      -> <bound method LifespanOn.receive of <uvicorn.lifespan.on.LifespanOn object at 0xffff4ff97b30>>\n          |    |        -> {'type': 'lifespan', 'asgi': {'version': '3.0', 'spec_version': '2.0'}, 'state': {}, 'app': <fastapi.applications.FastAPI obj...\n          |    -> <function Router.lifespan at 0xffff7f8ec0e0>\n          -> <fastapi.routing.APIRouter object at 0xffff50ab6e40>\n  File \"/usr/local/lib/python3.12/site-packages/starlette/routing.py\", line 693, in lifespan\n    async with self.lifespan_context(app) as maybe_state:\n               |    |                -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n               |    -> <function add_pagination.<locals>.lifespan at 0xffff50070860>\n               -> <fastapi.routing.APIRouter object at 0xffff50ab6e40>\n  File \"/usr/local/lib/python3.12/contextlib.py\", line 210, in __aenter__\n    return await anext(self.gen)\n                       |    -> <async_generator object add_pagination.<locals>.lifespan at 0xffff7cac8940>\n                       -> <contextlib._AsyncGeneratorContextManager object at 0xffff4ffe4c50>\n  File \"/usr/local/lib/python3.12/site-packages/fastapi_pagination/api.py\", line 414, in lifespan\n    async with _original_lifespan_context(app) as maybe_state:\n               |                          -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n               -> <function _merge_lifespan_context.<locals>.merged_lifespan at 0xffff501385e0>\n  File \"/usr/local/lib/python3.12/contextlib.py\", line 210, in __aenter__\n    return await anext(self.gen)\n                       |    -> <async_generator object _merge_lifespan_context.<locals>.merged_lifespan at 0xffff7ca43de0>\n                       -> <contextlib._AsyncGeneratorContextManager object at 0xffff4ffe52e0>\n  File \"/usr/local/lib/python3.12/site-packages/fastapi/routing.py\", line 133, in merged_lifespan\n    async with original_context(app) as maybe_original_state:\n               |                -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n               -> <function _merge_lifespan_context.<locals>.merged_lifespan at 0xffff500739c0>\n  File \"/usr/local/lib/python3.12/contextlib.py\", line 210, in __aenter__\n    return await anext(self.gen)\n                       |    -> <async_generator object _merge_lifespan_context.<locals>.merged_lifespan at 0xffff7ca43ef0>\n                       -> <contextlib._AsyncGeneratorContextManager object at 0xffff4ffe48c0>\n  File \"/usr/local/lib/python3.12/site-packages/fastapi/routing.py\", line 133, in merged_lifespan\n    async with original_context(app) as maybe_original_state:\n               |                -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n               -> <function _merge_lifespan_context.<locals>.merged_lifespan at 0xffff50070720>\n  File \"/usr/local/lib/python3.12/contextlib.py\", line 210, in __aenter__\n    return await anext(self.gen)\n                       |    -> <async_generator object _merge_lifespan_context.<locals>.merged_lifespan at 0xffff7cac4040>\n                       -> <contextlib._AsyncGeneratorContextManager object at 0xffff50001c70>\n  File \"/usr/local/lib/python3.12/site-packages/fastapi/routing.py\", line 133, in merged_lifespan\n    async with original_context(app) as maybe_original_state:\n               |                -> <fastapi.applications.FastAPI object at 0xffff50ab61b0>\n               -> <function get_lifespan.<locals>.lifespan at 0xffff50aa05e0>\n  File \"/usr/local/lib/python3.12/contextlib.py\", line 210, in __aenter__\n    return await anext(self.gen)\n                       |    -> <async_generator object get_lifespan.<locals>.lifespan at 0xffff7ca91930>\n                       -> <contextlib._AsyncGeneratorContextManager object at 0xffff50001ca0>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/main.py\", line 121, in lifespan\n    await initialize_services(fix_migration=fix_migration)\n          |                                 -> False\n          -> <function initialize_services at 0xffff52edf060>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/services/utils.py\", line 234, in initialize_services\n    await initialize_database(fix_migration=fix_migration)\n          |                                 -> False\n          -> <function initialize_database at 0xffff53c21440>\n> File \"/usr/local/lib/python3.12/site-packages/langflow/services/database/utils.py\", line 70, in initialize_database\n    await database_service.create_db_and_tables()\n          |                -> <function DatabaseService.create_db_and_tables at 0xffff52edfec0>\n          -> <langflow.services.database.service.DatabaseService object at 0xffff50002000>\n  File \"/usr/local/lib/python3.12/site-packages/langflow/services/database/service.py\", line 434, in create_db_and_tables\n    async with self.with_session() as session, session.bind.connect() as conn:\n               |    |                 |        |       |    -> <function AsyncEngine.connect at 0xffff7f2bdbc0>\n               |    |                 |        |       -> <sqlalchemy.ext.asyncio.engine.AsyncEngine object at 0xffff7ca7f990>\n               |    |                 |        -> <sqlmodel.ext.asyncio.session.AsyncSession object at 0xffff4ffe4800>\n               |    |                 -> <sqlmodel.ext.asyncio.session.AsyncSession object at 0xffff4ffe4800>\n               |    -> <function DatabaseService.with_session at 0xffff52edf740>\n               -> <langflow.services.database.service.DatabaseService object at 0xffff50002000>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/ext/asyncio/base.py\", line 121, in __aenter__\n    return await self.start(is_ctxmanager=True)\n                 |    -> <function AsyncConnection.start at 0xffff7f64b6a0>\n                 -> <sqlalchemy.ext.asyncio.engine.AsyncConnection object at 0xffff7ca603c0>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/ext/asyncio/engine.py\", line 274, in start\n    await greenlet_spawn(self.sync_engine.connect)\n          |              |    -> <member 'sync_engine' of 'AsyncConnection' objects>\n          |              -> <sqlalchemy.ext.asyncio.engine.AsyncConnection object at 0xffff7ca603c0>\n          -> <function greenlet_spawn at 0xffff81393380>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py\", line 201, in greenlet_spawn\n    result = context.throw(*sys.exc_info())\n             |       |      |   -> <built-in function exc_info>\n             |       |      -> <module 'sys' (built-in)>\n             |       -> <method 'throw' of 'greenlet.greenlet' objects>\n             -> <_AsyncIoGreenlet object at 0xffff7ca7f400 (otid=0xffff813cb540) dead>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 3274, in connect\n    return self._connection_cls(self)\n           |    |               -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n           |    -> <class 'sqlalchemy.engine.base.Connection'>\n           -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 148, in __init__\n    Connection._handle_dbapi_exception_noconnection(\n    |          -> <classmethod(<function Connection._handle_dbapi_exception_noconnection at 0xffff807a3d80>)>\n    -> <class 'sqlalchemy.engine.base.Connection'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 2439, in _handle_dbapi_exception_noconnection\n    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e\n          |                    |              |                 -> OperationalError('unable to open database file')\n          |                    |              -> (<class 'sqlite3.OperationalError'>, OperationalError('unable to open database file'), <traceback object at 0xffff7ca7ef80>)\n          |                    -> <method 'with_traceback' of 'BaseException' objects>\n          -> OperationalError('(sqlite3.OperationalError) unable to open database file')\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 146, in __init__\n    self._dbapi_connection = engine.raw_connection()\n    |                        |      -> <function Engine.raw_connection at 0xffff8079e5c0>\n    |                        -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n    -> <sqlalchemy.engine.base.Connection object at 0xffff4ffe4650>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/base.py\", line 3298, in raw_connection\n    return self.pool.connect()\n           |    |    -> <function Pool.connect at 0xffff80e028e0>\n           |    -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           -> Engine(sqlite+aiosqlite:////usr/local/lib/python3.12/site-packages/langflow/langflow.db)\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 449, in connect\n    return _ConnectionFairy._checkout(self)\n           |                |         -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           |                -> <classmethod(<function _ConnectionFairy._checkout at 0xffff80df8a40>)>\n           -> <class 'sqlalchemy.pool.base._ConnectionFairy'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 1263, in _checkout\n    fairy = _ConnectionRecord.checkout(pool)\n            |                 |        -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n            |                 -> <classmethod(<function _ConnectionRecord.checkout at 0xffff80e03560>)>\n            -> <class 'sqlalchemy.pool.base._ConnectionRecord'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 712, in checkout\n    rec = pool._do_get()\n          |    -> <function NullPool._do_get at 0xffff80e48860>\n          -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/impl.py\", line 308, in _do_get\n    return self._create_connection()\n           |    -> <function Pool._create_connection at 0xffff80e02660>\n           -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 390, in _create_connection\n    return _ConnectionRecord(self)\n           |                 -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n           -> <class 'sqlalchemy.pool.base._ConnectionRecord'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 674, in __init__\n    self.__connect()\n    -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 900, in __connect\n    with util.safe_reraise():\n         |    -> <class 'sqlalchemy.util.langhelpers.safe_reraise'>\n         -> <module 'sqlalchemy.util' from '/usr/local/lib/python3.12/site-packages/sqlalchemy/util/__init__.py'>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/langhelpers.py\", line 146, in __exit__\n    raise exc_value.with_traceback(exc_tb)\n          |         |              -> <traceback object at 0xffff7ca7f040>\n          |         -> <method 'with_traceback' of 'BaseException' objects>\n          -> OperationalError('unable to open database file')\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/pool/base.py\", line 896, in __connect\n    self.dbapi_connection = connection = pool._invoke_creator(self)\n    |    |                               |    |               -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n    |    |                               |    -> <function create_engine.<locals>.connect at 0xffff7cab42c0>\n    |    |                               -> <sqlalchemy.pool.impl.NullPool object at 0xffff50002600>\n    |    -> <member 'dbapi_connection' of '_ConnectionRecord' objects>\n    -> <sqlalchemy.pool.base._ConnectionRecord object at 0xffff7ca64b90>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/create.py\", line 646, in connect\n    return dialect.connect(*cargs, **cparams)\n           |       |        |        -> {'check_same_thread': False, 'timeout': 20}\n           |       |        -> ['/usr/local/lib/python3.12/site-packages/langflow/langflow.db']\n           |       -> <function DefaultDialect.connect at 0xffff809265c0>\n           -> <sqlalchemy.dialects.sqlite.aiosqlite.SQLiteDialect_aiosqlite object at 0xffff50afa600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/engine/default.py\", line 622, in connect\n    return self.loaded_dbapi.connect(*cargs, **cparams)\n           |    |            |        |        -> {'check_same_thread': False, 'timeout': 20}\n           |    |            |        -> ('/usr/local/lib/python3.12/site-packages/langflow/langflow.db',)\n           |    |            -> <function AsyncAdapt_aiosqlite_dbapi.connect at 0xffff53e918a0>\n           |    -> <sqlalchemy.dialects.sqlite.aiosqlite.AsyncAdapt_aiosqlite_dbapi object at 0xffff512424b0>\n           -> <sqlalchemy.dialects.sqlite.aiosqlite.SQLiteDialect_aiosqlite object at 0xffff50afa600>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/dialects/sqlite/aiosqlite.py\", line 355, in connect\n    await_only(connection),\n    |          -> <Connection(Thread-1, stopped daemon 281472138539392)>\n    -> <function await_only at 0xffff813931a0>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py\", line 132, in await_only\n    return current.parent.switch(awaitable)  # type: ignore[no-any-return,attr-defined] # noqa: E501\n           |       |             -> <Connection(Thread-1, stopped daemon 281472138539392)>\n           |       -> <attribute 'parent' of 'greenlet.greenlet' objects>\n           -> <_AsyncIoGreenlet object at 0xffff7ca7f400 (otid=0xffff813cb540) dead>\n  File \"/usr/local/lib/python3.12/site-packages/sqlalchemy/util/_concurrency_py3k.py\", line 196, in greenlet_spawn\n    value = await result\n                  -> <Connection(Thread-1, stopped daemon 281472138539392)>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 140, in _connect\n    self._connection = await future\n    |    |                   -> <Future finished exception=OperationalError('unable to open database file')>\n    |    -> None\n    -> <Connection(Thread-1, stopped daemon 281472138539392)>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 115, in run\n    result = function()\n             -> <function connect.<locals>.connector at 0xffff7ca58720>\n  File \"/usr/local/lib/python3.12/site-packages/aiosqlite/core.py\", line 392, in connector\n    return sqlite3.connect(loc, **kwargs)\n           |       |       |      -> {'check_same_thread': False, 'timeout': 20}\n           |       |       -> '/usr/local/lib/python3.12/site-packages/langflow/langflow.db'\n           |       -> <built-in function connect>\n           -> <module 'sqlite3' from '/usr/local/lib/python3.12/sqlite3/__init__.py'>\n\nsqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file\n(Background on this error at: https://sqlalche.me/e/20/e3q8)\n", "record": {"elapsed": {"repr": "0:03:29.396733", "seconds": 209.396733}, "exception": {"type": "OperationalError", "value": "(sqlite3.OperationalError) unable to open database file\n(Background on this error at: https://sqlalche.me/e/20/e3q8)", "traceback": true}, "extra": {}, "file": {"name": "utils.py", "path": "/usr/local/lib/python3.12/site-packages/langflow/services/database/utils.py"}, "function": "initialize_database", "level": {"icon": "❌", "name": "ERROR", "no": 40}, "line": 76, "message": "Error creating DB and tables", "module": "utils", "name": "langflow.services.database.utils", "process": {"id": 354, "name": "MainProcess"}, "thread": {"id": 281472896938016, "name": "MainThread"}, "time": {"repr": "2025-02-03 02:54:00.221118+00:00", "timestamp": 1738551240.221118}}}
